hangman/lib/hangman/game.ex

```diff
@@ -0,0 +1,128 @@
+defmodule Hangman.Game do
+
+  defmodule State do
+
+    defstruct(
+      game_state:    :initializing,
+      turns_left:    7,
+      answer:        "",
+      letters:       [],
+      used:          [],
+      last_guess:    ""
+    )
+
+  end
+
+  def new_game() do
+    word = Dictionary.random_word()
+    %State{answer: word, letters: fill_in_word(word, [])}
+  end
+
+  def tally(game) do
+    %{
+      game_state:     game.game_state,
+      turns_left:     game.turns_left,
+      letters:        game.letters,
+      used:           game.used,
+      last_guess:     game.last_guess
+    }
+  end
+
+  def make_move(game, guess) do
+
+    handle_guess(game, guess, game.last_guess,
+      first_matching_index(game.answer, guess),
+      first_blank_index(fill_in_word(game.answer, game.used ++ [ guess ])),
+      game.turns_left)
+
+  end
+
+ # Player loses
+  defp handle_guess(game, guess, _, nil, _, 1) do
+    new_state =
+      %State{game | game_state:     :lost,
+                    turns_left:     0,
+                    letters:        String.codepoints(game.answer),
+                    used:           game.used ++ [ guess ],
+                    last_guess:     guess
+      }
+
+    {new_state, tally(new_state)}
+
+  end
+
+  # Player's guess is a repeat
+  defp handle_guess(game, guess, guess, _, _, _) do
```

Comment:
## -3

`:already_used` is supposed to mean that the guess has already been guessed in this game, and not just that it is the same as the immediately previous guess.